compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmSharedVariablesManager.kt

```diff
@@ -146,6 +147,32 @@ class JvmSharedVariablesManager(
             val receiver = IrGetValueImpl(startOffset, endOffset, sharedVariableSymbol)
             IrSetFieldImpl(startOffset, endOffset, provider.elementField.symbol, receiver, unboxedValue, type, origin)
         }
+
+    fun getIrType(originalType: IrType): IrType {
+        val valueType = originalType
+        val provider = getProvider(InlineClassAbi.unboxType(valueType) ?: valueType)
+        val typeArguments = provider.refClass.typeParameters.map { valueType }
+        return provider.refClass.typeWith(typeArguments)
+    }
+
+    fun getSharedValue(sharedVariableSymbol: IrValueSymbol, originalGet: IrGetValue): IrExpression =
```

Comment:
Can we share this code with the other getSharedValue and setSharedValue? I believe there is an inheritance relationship between IrVariableSymbol and IrValueSymbol. Can one of the methods here just forward to the other?

Reply:
Yes, done; left over from my first drive through to getting it working; forgot to make it pretty :) 

I bumped the type on the interface `getSharedValue` to `IrValueSymbol`, which means I had to revise the JS and WASM implementations, too. I have run everything I could find locally involving JS and WASM and it seems to go fine, but we should see about a TeamCity build, I guess!

Reply:
TeamCity is happy :+1: